Epam/Java Basics_UpSkill Lab #1/Strings_and_basics_of_text_processing/Working_with_a_string_as_a_String_or_StringBuilder/src/com/androsor/string_object/HomeTask2.java

```diff
@@ -0,0 +1,42 @@
+package com.androsor.string_object;
+
+import java.util.Scanner;
+
+/**
+ * In the line insert after each 'a' character 'b'.
+ */
+
+public class HomeTask2 {
+
+    public static void main(String[] args) {
+
+        String stringReplaceable;
+        char symbolA;
+        char symbolB;
+
+        stringReplaceable = HomeTask1.enterFromConsole("Введите строку:");
+        symbolA = enterCharFromConsole("Введите заменяемый символ:");
+        symbolB = enterCharFromConsole("Введите заменяющий символ:");
+
+        StringBuilder newString = new StringBuilder(stringReplaceable);
+
+        for ( int i = 0; i < newString.length(); i++) {
+
+            if (newString.charAt(i) == symbolA) {
+                newString.insert(i + 1, symbolB);
+            }
+        }
+
+        System.out.printf("Измененная строка:\n %s", newString);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.